src/integration/java/io/pinecone/helpers/TestIndexResourcesManager.java

```diff
@@ -122,7 +122,7 @@ private String createOrGetPodIndex() throws InterruptedException, PineconeExcept
         waitUntilIndexIsReady(pineconeClient, indexName);
 
         // Explicitly wait after ready to avoid the "no healthy upstream" issue
-        Thread.sleep(10000);
+        Thread.sleep(30000);
```

Comment:
(Not rly related to this PR, but...)

Reply:
I thought about that, from what I've seen from the "no healthy upstream" failures I think we only see them on pod indexes, and I think it's because pod indexes take a bit longer to be fully ready. I didn't want to increase the wait for all indexes just pod indexes.